sdk/dynamicconfig/metric/controller/push/period_matcher.go

```diff
@@ -51,26 +52,46 @@ func (matcher *PeriodMatcher) MarkStart(startTime time.Time) {
 // ApplySchedules sets the schedules that a PeriodMatcher consults when
 // constructing a Rule. After processing the schedules, ApplySchedules returns
 // the optimal period with which a controller should run a collection sweep.
+// If a period of 0 is returned, then metric collection should be halted.
 //
 // This function may be called concurrently.
-func (matcher *PeriodMatcher) ApplySchedules(sched []*pb.MetricConfigResponse_Schedule) time.Duration {
+func (matcher *PeriodMatcher) ApplySchedules(sched []*pb.MetricConfigResponse_Schedule) (time.Duration, error) {
+	if err := validate(sched); err != nil {
+		return 0, err
+	}
+
 	matcher.m.Lock()
 	matcher.sched = sched
 	matcher.metrics = make(map[string]*collectData)
 	matcher.m.Unlock()
 
-	return getExportPeriod(matcher.sched)
+	return getExportPeriod(matcher.sched), nil
+}
+
+func validate(sched []*pb.MetricConfigResponse_Schedule) error {
+	if len(sched) == 0 {
+		return errors.New("no schedules")
+	}
+
+	for _, schedule := range sched {
+		if schedule.PeriodSec < 0 {
```

Comment:
"<= 0"

I think we decided not to support 0 period in the protocol, right?

Reply:
I believe that period=0 mean to not export, for example: https://github.com/open-telemetry/opentelemetry-proto/blob/6232d3382db3bb098d0a0e7aeb667a50f1ffc1fc/opentelemetry/proto/experimental/metrics/configservice/configservice.proto#L86

Reply:
Okay, then, let's update the error messae to "must not be negative".